app/controllers/daily_reports_controller.rb

```diff
@@ -0,0 +1,120 @@
+# frozen_string_literal: true
+class DailyReportsController < InheritedResources::Base
+  before_action :assign_saved_report, only: [:edit, :update]
+
+  def index
+    @q = DailyReport.includes(user: :groups).ransack(search_params)
+    @daily_reports = @q.result(distinct: true).released.order('shipped_at desc').page params[:page]
+  end
+
+  def user
+    @target_year = (params[:target_year] || Date.today.year).to_i
+    @target_month = (params[:target_month] || Date.today.month).to_i
+    @report_user = User.find(params[:user_id])
+    @daily_reports = user_reports_in_year_month(@target_year, @target_month, @report_user)
+  end
+
+  def show
+    @daily_report = DailyReport.includes(comments: { user: :user_profile }).find(params[:id])
+    raise(Forbidden, 'can not see wip reports of other users') unless @daily_report.browseable?(current_user)
+  end
+
+  def new
+    target_date = params[:target_date] || Date.today
+    @daily_report = current_user.daily_reports.build(target_date: target_date)
+  end
+
+  def create
+    @daily_report = DailyReport.new(permitted_params.merge!(user: current_user))
+    @daily_report.shipped_at ||= Time.current unless params[:wip]
+    shipped_at_was = @daily_report.shipped_at_was
+    if @daily_report.save
+      daily_report_notify(shipped_at_was)
+      redirect_to @daily_report
+    else
+      flash_errors(@daily_report)
+      render :new
+    end
+  end
+
+  def edit; end
+
+  def update
+    @daily_report.assign_attributes(permitted_params)
+    @daily_report.shipped_at ||= Time.current unless params[:wip]
+    shipped_at_was = @daily_report.shipped_at_was
+
+    if @daily_report.save
+      daily_report_notify(shipped_at_was)
+      redirect_to @daily_report
+    else
+      flash_errors(@daily_reports)
+      render :edit
+    end
+  end
+
+  def copy
+    @daily_report = DailyReport.new(target_date: params[:target_date], user: current_user).set_prev_daily_report
+    render :new
+  end
+
+  private
+
+  def assign_saved_report
+    @daily_report = current_user.daily_reports.find params[:id]
+    @saved_shipped_at = @daily_report.shipped_at
+  end
+
+  def flash_errors(daily_report)
+    flash.now[:error] = daily_report.errors.full_messages
+  end
+
+  def user_reports_in_year_month(year, month, report_user)
```

Comment:
__[reek]__
* [UtilityFunction] doesn't depend on instance state (maybe move it to another class?) [:ref](https://github.com/troessner/reek/blob/master/docs/Utility-Function.md)

<!-- Comment by [SideCI](https://sideci.com) -->